progs/st/main.go

```diff
@@ -114,17 +114,27 @@ func newNode(startIndex int, endIndex int, parent *Node) *Node {
 /* creates a new internal node between parent and child.
 it also creates a new node branching (head)*/
 func splitEdge(parent *Node, child *Node, splitIndex int, start_idx int) {
+	fmt.Println("splitting edge")
 	delete(parent.Children, rune(x[child.startIndex]))
 
 	new_internal := newNode(child.startIndex, child.startIndex+splitIndex, parent)
-	newNode(start_idx, len(x), new_internal)
+	fmt.Println(start_idx+splitIndex, start_idx, splitIndex)
+	s := newNode(start_idx, len(x), new_internal)
 
 	child.startIndex = child.startIndex + splitIndex
+
 	new_internal.Children[rune(x[child.startIndex])] = child
 
+	fmt.Println("split:", parent, child, new_internal, s)
+
 }
 
-func findoccurrence(root *Node, y string) { return }
+func findoccurrences(root *Node, y string) []int_tuple {
+	parent_tree, _, _, _ := search(root, y)
+	fmt.Println(parent_tree, "cringe")
+	return BfOrder(parent_tree)
+
+}
```

Comment:
If I read the code correctly, `search(root, y)` will return the location where `y` is in the tree *if it is there*, or the location of the longest path that matches a *prefix* of `y`. If it is the former, then we can use it to get the occurrences of `y` in `x` (via any traversal that gives us the leaves). If the latter, there shouldn't be any occurrences. Knowing `parent_tree` doesn't tell me if `y` matched or not. So, shouldn't you test that before you call `BfOrder()`?

Also, why would occurrences be reported as integer pairs? An occurrence is either an index (for exact pattern matching) or an index and a CIGAR (for approximative matching).

Ah, I see that the traversal returns. That is not what we want at all. You just want the labels in all the leaves in the subtree.

Reply:
Ah very true about search... We should get that fixed.
The integer pairs were just some quick thing we used while debugging... Those will be replaced B)